KomaMRIBase/src/motion/motionlist/actions/arbitraryactions/FlowPath.jl

```diff
@@ -0,0 +1,37 @@
+@doc raw"""
+    flowpath = FlowPath(dx, dy, dz, spin_reset)
+
+FlowPath struct. This action is the same as `Path`, 
+except that it includes an additional field, called `spin_reset`, 
+which accounts for spins leaving the volume and being remapped 
+to another input position. When this happens, the magnetization 
+state of these spins must be reset during the simulation. 
+
+As with the `dx`, `dy` and `dz` matrices, `spin_reset`
+has a size of (``N_{spins} \times \; N_{discrete\,times}``).
+
+# Arguments
+- `dx`: (`::AbstractArray{T<:Real}`, `[m]`) displacements in x
+- `dy`: (`::AbstractArray{T<:Real}`, `[m]`) displacements in y 
+- `dz`: (`::AbstractArray{T<:Real}`, `[m]`) displacements in z
+- `spin_reset`: (`::AbstractArray{T<:Real}`) reset spin state flags
+
+# Returns
+- `flowpath`: (`::FlowPath`) FlowPath struct
+
+# Examples
+```julia-repl
+julia> flowpath = FlowPath(
+           dx=[0.01 0.02; 0.02 0.03], 
+           dy=[0.02 0.03; 0.03 0.04], 
+           dz=[0.03 0.04; 0.04 -0.04],
+           spin_reset=[false false; false true]
+       )
+```
+"""
+@with_kw struct FlowPath{T<:Real} <: ArbitraryAction{T}
+    dx::AbstractArray{T}
+    dy::AbstractArray{T}
+    dz::AbstractArray{T}
+    spin_reset::AbstractArray{T}
```

Comment:
I have made some modifications to ArbitraryAction.jl to allow the coefficients and nodes to have different eltypes. It works on CPU and CUDA but I have to check in oneAPI because I don't trust it.

Reply:
Thanks for looking into this, let's keep it like this, maybe add a TODO or open an issue to remember to change this at some point.

Reply:
I open an issue so I don´t make more commits haha

Reply:
Chiming in since I have an idea on how to fix the oneAPI cumsum kernel. Does this work?

```julia
@kernel function naive_cumsum!(B, @Const(A))
    i = @index(Global)
    T = eltype(A)

    cur_val = zero(T)
    for k ∈ 1:size(A, 2)
        @inbounds cur_val += A[i, k]
        @inbounds B[i, k] = cur_val
    end
end
```

Reply:
It works! Thank you, I have committed that change in [5780c31](https://github.com/JuliaHealth/KomaMRI.jl/pull/442/commits/5780c3102174e430393cb4d082fed23ab48b4b60)